src/main/java/org/radarcns/webapp/resource/AggregatedDataPointsEndPoint.java

```diff
@@ -0,0 +1,98 @@
+import org.radarcns.domain.restapi.TimeWindow;
+import org.radarcns.domain.restapi.dataset.AggregatedDataPoints;
+import org.radarcns.domain.restapi.header.TimeFrame;
+import org.radarcns.listener.managementportal.ManagementPortalClient;
+import org.radarcns.service.DataSetService;
+import org.radarcns.webapp.filter.Authenticated;
+import org.radarcns.webapp.param.DataAggregateParam;
+import org.radarcns.webapp.param.InstantParam;
+import org.radarcns.webapp.validation.Alphanumeric;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Authenticated
+@Path("/" + AGGREGATED_DATA_POINTS)
+public class AggregatedDataPointsEndPoint {
+
+    private static final Logger LOGGER = LoggerFactory
+            .getLogger(AggregatedDataPointsEndPoint.class);
+
+    @Inject
+    private ManagementPortalClient mpClient;
+
+    @Inject
+    private DataSetService dataSetService;
+
+    //--------------------------------------------------------------------------------------------//
+    //                                    Aggregated Data Points API functions                    //
+    //--------------------------------------------------------------------------------------------//
+
+    /**
+     * JSON function that returns aggregated volumes of source-data requested.
+     */
+    @POST
+    @Produces({APPLICATION_JSON, AVRO_BINARY})
+    @Consumes({APPLICATION_JSON})
+    @Path("/{" + PROJECT_NAME + "}/{" + SUBJECT_ID + "}")
+    @Operation(summary = "Returns a Dataset object formatted in JSON.",
+            description = "Each collected sample is aggregated to provide near real-time "
+                    + "statistical results. This end-point returns the latest available record "
+                    + "for the stat for the given projectID, subjectID, sourceID and SourceDataName"
+                    + " Data can be queried using different time-frame resolutions.")
+    @ApiResponse(responseCode = "500", description = "An error occurs while executing")
+    @ApiResponse(responseCode = "200", description = "Returns a dataset object containing latest "
+            + "available record for the given inputs")
+    @ApiResponse(responseCode = "401", description = "Access denied error occurred")
+    @ApiResponse(responseCode = "403", description = "Not Authorised error occurred")
+    @ApiResponse(responseCode = "404", description = "Subject not found.")
+    @NeedsPermissionOnSubject(entity = MEASUREMENT, operation = READ)
+    public AggregatedDataPoints getAggregatedDataPoints(DataAggregateParam aggregateParam,
+            @Alphanumeric @PathParam(PROJECT_NAME) String projectName,
+            @Alphanumeric @PathParam(SUBJECT_ID) String subjectId,
+            @QueryParam(TIME_WINDOW) TimeWindow interval,
+            @QueryParam(START) InstantParam start,
+            @QueryParam(END) InstantParam end) throws IOException {
+
+        mpClient.getProject(projectName);
+        mpClient.checkSubjectInProject(projectName, subjectId);
```

Comment:
why add two calls here? If the subject is in the project, apparently the project exists, right?

Reply:
Ideally yes, given that we also don't `delete` any projects which collected data.

Reply:
It thought it wouldn't hurt because of the cache.